src/server/handlers/consents/{ID}/generateChallenge.ts

```diff
@@ -0,0 +1,110 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files
+ are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+
+ --------------
+ ******/
+// eslint-disable-next-line max-len
+import { updateConsentCredential, isConsentRequestInitiatedByValidSource, putConsentId, ConsentCredential } from '../../../../domain/consents/{ID}/generateChallenge'
+import * as challenge from '../../../../lib/challenge'
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import { consentDB, scopeDB } from '../../../../lib/db'
+import { Consent } from '../../../../model/consent'
+import Logger from '@mojaloop/central-services-logger'
+import { convertScopesToExternal } from '../../../../lib/scopes'
+import { Scope } from '../../../../model/scope'
+
+// Asynchronously deals with generating challenge, updating consent db
+//  and making outgoing PUT consent/{ID} call
+export async function postBackground (
```

Comment:
Where did the naming convention on these come from? Is it a hapi standard or are we choosing them? They're way less descriptive than your other method names which are much clearer.

Reply:
So @eoln  and I were working on this together - this is what used to be within post() within a setImmediate call - not completely sure if this needs to be more descriptive as it is essentially just the 'background/async' section of the post handler?

Reply:
you are right @jgeewax, the name is ugly